components/builder-jobsrv/src/server/log.rs

```diff
@@ -0,0 +1,44 @@
+use error::Result;
+use std::fs;
+use std::path::PathBuf;
+
+/// Encapsulates the local filesystem directory in which in-process
+/// build job logs will be collected prior to being sent to long-term
+/// storage.
+#[derive(Clone, Debug)]
+pub struct LogDirectory {
```

Comment:
This is called the [newtype pattern](https://aturon.github.io/features/types/newtype.html) in Rust. It's recommended to represent this pattern as a [tuple struct](https://doc.rust-lang.org/book/structs.html#tuple-structs) with one element. You can later acces that element in your functions as `self.0`